File: src/leetcode/2026/04/Solution724.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution724 {
public:
    int pivotIndex(vector<int>& nums) {
        int total = 0;
        for (int num : nums) {
            total += num;
        }

        int leftSum = 0;
        for (int i = 0; i < nums.size(); ++i) {
            if (leftSum == total - leftSum - nums[i]) {
                return i;
            }
            leftSum += nums[i];
        }

        return -1;
    }
};

int main() {
    Solution724 solution;

    vector<int> nums = {1, 7, 3, 6, 5, 6};
    int pivotIndex = solution.pivotIndex(nums);
    // Output: 3
    cout << "Pivot Index: " << pivotIndex << endl;

    nums = {1, 2, 3};
    pivotIndex = solution.pivotIndex(nums);
    // Output: -1
    cout << "Pivot Index: " << pivotIndex << endl;

    nums = {2, 1, -1};
    pivotIndex = solution.pivotIndex(nums);
    // Output: 0
    cout << "Pivot Index: " << pivotIndex << endl;
    
    return 0;
}

File: src/leetcode/2026/04/Solution724.java

public class Solution724 {
    public int pivotIndex(int[] nums) {
        int totalSum = 0;
        for (int num : nums) {
            totalSum += num;
        }

        int leftSum = 0;
        for (int i = 0; i < nums.length; i++) {
            if (leftSum == totalSum - leftSum - nums[i]) {
                return i; // Found the pivot index
            }
            leftSum += nums[i];
        }

        return -1; // No pivot index found
    }

    public static void main(String[] args) {
        Solution724 solution = new Solution724();

        int[] input = { 1, 7, 3, 6, 5, 6 };
        int output = solution.pivotIndex(input);
        System.out.println("Output: " + output);

        input = new int[] { 1, 2, 3 };
        output = solution.pivotIndex(input);
        System.out.println("Output: " + output);

        input = new int[] { 2, 1, -1 };
        output = solution.pivotIndex(input);
        System.out.println("Output: " + output);
    }
}
